app/workers/worker_base.rb

```diff
@@ -0,0 +1,7 @@
+class WorkerBase
```

Comment:
I don't see that this makes enough of a difference to justify changing what has already been done. As you say, there are lots of ways to achieve the same thing in Ruby - and this seems more verbose for little benefit.

Let's stick with the base class.